packages/babel-parser/src/parser/expression.ts

```diff
@@ -1435,7 +1451,13 @@ export default class ExpressionParser extends LValParser {
     // If the pipe-operator plugin’s configuration matches the current token’s type,
     // then this will return `node`, will have been finished as a topic reference.
     // Otherwise, this will throw a `PipeTopicUnconfiguredToken` error.
-    return this.finishTopicReference(node, startLoc, pipeProposal, tokenType);
+    return this.finishTopicReference(
+      // @ts-expect-error using "Undone" node as "done"
```

Comment:
Can you update `finishTopicReference`? It should only work with `Undone<N.TopicReference>`.